csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -250,6 +250,10 @@ module LocalFlow {
         scope = e2 and
         isSuccessor = true
         or
+        e1 = e2.(CollectionExpression).getAnElement().(SpreadElementExpr).getExpr() and
```

Comment:
I think it is better to break this up into two steps:

For `[..x]`, add a local flow step from `x` to `..x`, and another local flow step from `..x` to `[..x]`. We can then implement `expectsContent` on the `..x` node, so that flow will only be allowed when we are in fact tracking flow inside a collection.

Reply:
Wow, that is a really nice catch @hvitved !!!